tools/osv-scanner/main.go

```diff
@@ -149,30 +157,79 @@ func scanDebianDocker(query *osv.BatchedQuery, dockerImageName string) {
 		if len(splitText) != 2 {
 			log.Fatalf("Unexpected output from Debian container: \n\n%s", text)
 		}
-		pkgDetails := osv.MakePkgRequest(lockfile.PackageDetails{
+		pkgDetailsQuery := osv.MakePkgRequest(lockfile.PackageDetails{
 			Name:    splitText[0],
 			Version: splitText[1],
 			// TODO(rexpan): Get and specify exact debian release version
 			Ecosystem: "Debian",
 		})
-		query.Queries = append(query.Queries, pkgDetails)
+		pkgDetailsQuery.Source = "docker:" + dockerImageName
+		query.Queries = append(query.Queries, pkgDetailsQuery)
 	}
 	log.Printf("Scanned docker image")
 }
 
+// printResults prints the osv scan results into a human friendly table.
 func printResults(query osv.BatchedQuery, resp *osv.BatchedResponse) {
+	outputTable := table.NewWriter()
+	outputTable.SetOutputMirror(os.Stdout)
+	outputTable.AppendHeader(table.Row{"Source", "Ecosystem", "Affected Package", "Installed Version", "Vulnerability ID", "OSV URL"})
+
+	max := func(a int, b int) int {
+		if a > b {
+			return a
+		} else {
+			return b
+		}
+	}
+	maxCharacters := 0
+
 	for i, query := range query.Queries {
 		if len(resp.Results[i].Vulns) == 0 {
 			continue
 		}
-
-		var urls []string
 		for _, vuln := range resp.Results[i].Vulns {
-			urls = append(urls, osv.BaseVulnerabilityURL+vuln.ID)
+			outputRow := table.Row{query.Source}
+			shouldMerge := false
+			if query.Commit != "" {
+				outputRow = append(outputRow, "GIT", query.Commit, query.Commit)
+				maxCharacters = max(maxCharacters, len(query.Commit))
+				shouldMerge = true
+			} else if query.Package.PURL != "" {
+				outputRow = append(outputRow, "PURL", query.Package.PURL, query.Package.PURL)
```

Comment:
Rather than duplicate the PURL, let's put it in a single column (or split it up into base PURL + version).

Reply:
So this actually will merge them into one, which looks like this:
![image](https://user-images.githubusercontent.com/106129829/187833479-7da01bb9-8ee4-4ac7-97ec-11b9b03b5792.png)
Well the duplicate commits will result in something like this, for PURLs I am now parsing it to split it up as suggested.

Reply:
Wow, that's pretty magical. That's fine for now then!

Reply:
Or if you did it already, let's just go with that. I think we eventually want to abstract away PURLS here and show a consistent "Ecosystem + Package + version"